src/TinyIoC/TinyIoC.cs

```diff
@@ -3580,53 +3580,54 @@ where localType.IsAssignableFrom(implementationType)
-                asm => asm.FullName.StartsWith("mscorlib,", StringComparison.Ordinal),
-                asm => asm.FullName.StartsWith("CR_VSTest", StringComparison.Ordinal),
-                asm => asm.FullName.StartsWith("DevExpress.CodeRush", StringComparison.Ordinal),
-                asm => asm.FullName.StartsWith("xunit.", StringComparison.Ordinal),
-            };
+            asm => asm.FullName.StartsWith("Microsoft.", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("System.", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("System,", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("CR_ExtUnitTest", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("mscorlib,", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("CR_VSTest", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("DevExpress.CodeRush", StringComparison.Ordinal),
+            asm => asm.FullName.StartsWith("xunit.", StringComparison.Ordinal),
+        };
 
-            foreach (var check in ignoreChecks)
+        private bool IsIgnoredAssembly(Assembly assembly)
+        {
+            for (int i = 0; i < ignoredAssemlies.Count; i++)
             {
-                if (check(assembly))
+                if (ignoredAssemlies[i].Invoke(assembly))
                     return true;
             }
 
             return false;
         }
-
-        private bool IsIgnoredType(Type type, Func<Type, bool> registrationPredicate)
+        
+        // TODO - find a better way to remove "system" types from the auto registration
+        private readonly List<Func<Type, bool>> ignoreChecks = new List<Func<Type, bool>>()
         {
-            // TODO - find a better way to remove "system" types from the auto registration
-            var ignoreChecks = new List<Func<Type, bool>>()
-            {
-                t => t.FullName.StartsWith("System.", StringComparison.Ordinal),
-                t => t.FullName.StartsWith("Microsoft.", StringComparison.Ordinal),
-                t => t.IsPrimitive(),
+            t => t.FullName.StartsWith("System.", StringComparison.Ordinal),
+            t => t.FullName.StartsWith("Microsoft.", StringComparison.Ordinal),
+            t => t.IsPrimitive(),
 #if !UNBOUND_GENERICS_GETCONSTRUCTORS
                 t => t.IsGenericTypeDefinition(),
 #endif
-                t => (t.GetConstructors(BindingFlags.Instance | BindingFlags.Public).Length == 0) && !(t.IsInterface() || t.IsAbstract()),
-            };
+            t => (t.GetConstructors(BindingFlags.Instance | BindingFlags.Public).Length == 0) && !(t.IsInterface() || t.IsAbstract()),
+        };
 
-            if (registrationPredicate != null)
+        private bool IsIgnoredType(Type type, Func<Type, bool> registrationPredicate)
+        {
+            if (registrationPredicate != null && !registrationPredicate(type))

```

Comment:
Seems like this should be stored in a temp variable and added to the ignore checks below (saved an extra lookup)

Reply:
This is done as you said or am I missing something?